db_upgrade_scripts/README.md

```diff
@@ -1 +1,63 @@
-Directory contains sql scripts to be executed for DB migrations.
\ No newline at end of file
+## MOSIP Database Upgrade and Rollback scripts inventory guidelines on postgresql database.
+## Prerequisities
+* Postgres client (psql) has to be installed on the deployment servers.
+* DB Server and access details.
+* Copy database upgrade scripts(DDL, DML, .SH ... etc) from git/repository on to the DB deployment server.
+* 
+* Necessary details needs to be updated in peoperties file against to the relevant variables being used 
+
+## Properties file variable details and description,Properties file has to be updated with the required details for each database before proceeding with upgrade or rollback steps.
+
+**ACTION=** Action to be taken, possible values: upgrade or rollback
+
+**MOSIP_DB_NAME=** Destination DB name to upgrade
+
+**DB_SERVERIP=** DB Server Ip
+
+**DB_PORT=** DB Server Port
+
+**SU_USER=** Superuser to connect to DB server
+
+**SU_USER_PWD=** Superuser password to connect to DB
+
+**DEFAULT_DB_NAME=** Default database name to connect with respective postgres server i.e. ex: postgres
+
+**DBUSER_PWD=** DB user password for required postgres DB
+
+**CURRENT_VERSION=** Current DB version
+
+**UPGRADE_VERSION=** final DB version after upgrade
+
+## DB upgrade/rollback deployment:
+**Step-1:** Make modification to all the respective database properties files **(<<schema>>_upgrade.properties)** in the respective database directories.
+
+**Step-2:** DB upgrade deployment for all databases, run the **"<<schema>>_upgrade.sh"** script with upgrade version as parameter. run below command
+
+```bash
+./upgrade.sh upgrade.properties
+```
+**or**
+```bash
+bash upgrade.sh upgrade.properties
+```
+**Note:** Observe the deployment and make sure DB should be updated
```

Comment:
## Upgrade / Rollback DB 

*  Make modifications to all the respective database properties files `<<schema>>_upgrade.properties` in the respective database directories.

*  Ensure to update the upgrade.properties file as per your requirement. Run the below command to `Upgrade` / `Rollback` operation.

```bash

cd db_upgrade_script/<db-script>/

./upgrade.sh upgrade.properties 
```

**Note:** 
* Examine the output generated by the upgrade scripts to verify the successful execution of the operation, ensuring it is free from any errors or anomalies.
* Accessing the right path for DB operation
* Ensure the DB server is accessible
* Loading data or DML operations are valid only for those DBs that carry DML actions.